packages/core/src/storage/sovranStorage.ts

```diff
@@ -0,0 +1,115 @@
+import { createStore, Store } from '@segment/sovran-react-native';
+import type {
+  SegmentAPIIntegrations,
+  IntegrationSettings,
+  SegmentEvent,
+  DeepPartial,
+  Context,
+  UserInfoState,
+} from '..';
+import type { Storage } from './types';
+
+type Data = {
+  isReady: boolean;
+  events: SegmentEvent[];
+  eventsToRetry: SegmentEvent[];
+  context: DeepPartial<Context>;
+  settings: SegmentAPIIntegrations;
+  userInfo: UserInfoState;
+};
+const INITIAL_VALUES: Data = {
+  isReady: true,
+  events: [],
+  eventsToRetry: [],
+  context: {},
+  settings: {},
+  userInfo: {
+    anonymousId: 'anonymousId',
+    userId: undefined,
+    traits: undefined,
+  },
+};
+
+export class SovranStorage implements Storage {
+  private storeId: string;
+  private contextStore: Store<DeepPartial<Context>>;
+  private settingsStore: Store<SegmentAPIIntegrations>;
+  private eventsStore: Store<SegmentEvent[]>;
+  private userInfoStore: Store<UserInfoState>;
+
+  constructor(storeId: string) {
+    this.storeId = storeId;
+    this.contextStore = createStore(INITIAL_VALUES.context, {
+      persist: { storeId: `${this.storeId}-context` },
+    });
+    this.settingsStore = createStore(INITIAL_VALUES.settings, {
+      persist: { storeId: `${this.storeId}-settings` },
+    });
+    this.eventsStore = createStore(INITIAL_VALUES.events, {
+      persist: { storeId: `${this.storeId}-events` },
+    });
+    this.userInfoStore = createStore(INITIAL_VALUES.userInfo, {
+      persist: { storeId: `${this.storeId}-userInfo` },
+    });
+  }
+
+  readonly isReady = {
+    get: () => true,
+    onChange: (_callback: (value: boolean) => void) => {
+      // Not doing anything cause we don't yet have a way to persist
```

Comment:
NIT we should remove this comment and maybe this function entirely?

Reply:
Let's not for now, I think it might actually come in handy if we want to do some preprocessing or anything during initialization